dotCMS/src/main/java/com/dotmarketing/quartz/job/FreeServerFromClusterJob.java

```diff
@@ -56,8 +57,13 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
 
                     if (shouldRewire) {
                         //Rewires the other nodes.
-                        ClusterFactory.addNodeToCacheCluster(APILocator.getServerAPI().getCurrentServer());
-                        ClusterFactory.addNodeToESCluster();
+                        if(ClusterUtils.isTransportAutoWire()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)